Add support for configuring a lookback and refetching data for the lookback window

mjn opened the pull request:
## What was the issue?

Currently, the TikTok tap loads data for each day at most one time. However, TikTok data may take up to 30 days to stabilize as conversion signals are sent back to TikTok.

Thus, if you are extracting data daily, the data that you pull down may not be complete.

## What was done?

In order to support refetching data to ensure that you have the most up to date metrics, this PR adds a new `lookback` configuration option that takes an integer as the value. This value specified the lookback window for which you want to reload data.

If `lookback` is configured, the tap will set the `start_date` to be the maximum between the configured `start_date` and the current date minus the number of lookback days (to ensure that it doesn't collect data prior to the configured `start_date`).

mjn committed:
Add support for configuring a lookback and refetching data for the lookback window

GtheSheep commented on tap_tiktok/streams.py:
```diff
@@ -255,6 +255,16 @@ def get_url_params(
+        if lookback_window > 0:
+            # if lookback is configured, we want to refetch data for the entire lookback window
+            # (or as far back as the configured start date, whichever is the most recent date)
+            start_date = max(
```
would this override the `start_date` from the pagination if `lookback` is set? i.e. Should this happen where we fetch the initial `start_date` on line 257? And then just be the bookmarked value minus the `lookback`

mjn replied:
Good catch! I have moved the lookback handling into the `else` block where the initial or bookmarked start date is loaded, so it will no longer override start date from pagination.

GtheSheep commented:
Thanks for catching this @mjn, appreciate it!!

mjn committed:
Only consider lookback when loading initial or bookmarked start date

mjn committed:
Document lookback configuration option

mjn commented:
I have updated the date parsing of `self.config["start_date"]` to use [`dateutil.parser.isoparse`](https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.isoparse) as it has better ISO format parsing when using python 3.9.

I have also updated the `README` to document the `lookback` configuration option.
